keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/ArcSinVertex.java

```diff
@@ -26,8 +25,17 @@ protected DoubleTensor op(DoubleTensor value) {
     }
 
     @Override
-    protected DualNumber dualOp(DualNumber dualNumber) {
-        return dualNumber.asin();
+    protected PartialDerivatives forwardModeAutoDifferentiation(PartialDerivatives derivativeOfParentWithRespectToInputs) {
+
+        DoubleTensor inputValue = inputVertex.getValue();
+
+        if (derivativeOfParentWithRespectToInputs.isEmpty()) {
```

Comment:
I was going to refactor it in another PR but we might as well clean it up here. This would require that the autodiff traverse never passes an empty partial to a vertex, which I think is already the case.